spec.md

```diff
@@ -523,13 +518,295 @@ schema being used.
         "bullets": [
           "40 concurrent connections"
         ]
+      },
+      "schemas": {
+        "service_instance": {
+          "create": {
+            "parameters": {
+              "$ref": "osb.v2:///fakeservice/v1#BillingAccountParameters"
+            }
+          },
+          "update": {
+            "parameters": {
+              "$schema": "http://json-schema.org/draft-04/schema#",
+              "type": "object",
+              "propeties": {
+                "some-other-account": {
+                  "description": "Other fake account needed for update.",
+                  "type": "string"
+                },
+                "billing": {
+                  "$ref": "osb.v2:///fakeservice/v1#BillingAccountParameters"
+                }
+              }
+            }
+          }
+        },
+        "service_binding": {
+          "create": {
+            "parameters": {
+              "$ref": "osb.v2:///fakeservice/v1/bindings#AccountParameters"
+            }
+          }
+        }
       }
     }]
   }]
 }
 ```
 
 
+## JSON Schemas
+
+The JSON Schemas endpoint is used to define reusable definitions of JSON Schema
+Objects referenced throughout the [Catalog](#catalog-management). It will be the
+responsibility of the Platform to examine the response from the catalog to
+determine if the Broker is using `$ref` JSON Schemas. The Broker can signal to
+the Platform that JSON Schema components are supported by using the `osb_v2:`
+scheme. Components are a way to allow Platforms to only fetch sub-schemas.
+
+The root JSON Schema document can be found by getting the JSON Schemas
+endpoint without a `:component_path`.
+
+If the `osb_v2:` scheme is used by the broker, then the Platform MAY fetch the
+component JSON Schema directly using the following convention:
+
+Given: `osb_v2:///{:component_path}`, fetch `/v2/catalog/schemas/:component_path`
+and then the Platform is responsible to continue fetching any unknown
+`:component_path` URIs. Care ought to be taken by the platform to not fetch
+duplicate definitions.
```

Comment:
Not sure what this means by fetching "duplicate definitions".

Reply:
Fetching the same `:component_path`. So then you end up with a list containing two copies of a schema with the same id.

Reply:
I'm still not following. Are you saying that if `osb_v2:///foo` appears twice in the schema, then the Platform only needs to retrieve it once? If so, yes I agree, but the "Care ought to be taken..." sentence makes it sound like something bad will happen if they do retrieve it more than once.  I would suggest that rather than making it sound like there's something people need to be worried about, we phrase it more like a good thing, an optimization. For example, talk about how this feature gives Platforms a more optimized schema processing and retrieval behavior by allowing portions of the schema that are used in multiple places to only be retrieved once. Or something like that.

Reply:
Maybe you can help me wordsmith this, the care comes into play if the platform chooses to implement the external schemas as a dumb loop that downloads and appends the schema to a list to be given to the pre-fetched schema list. If that document collection has objects that collide in id, the processing library is unable to process any ids. That is the warning I am trying to put here. 

Maybe ... "Platforms MUST read the JSON Schema spec." ?